### PR TITLE
feat: conf inherit

### PR DESCRIPTION
**Pull request by kevwan**

### CODE REVIEW

File: core/mapping/unmarshaler.go [hunk 1]
```diff
@@ -353,16 +350,29 @@ func (u *Unmarshaler) processNamedField(field reflect.StructField, value reflect
...
+			parent: valuer,
+		}, key, opts, fullName)
 	}
 
-	return u.processNamedFieldWithoutValue(field, value, opts, fullName)
+	return u.processNamedFieldWithValue(field, value, valueWithParent{
+		value:  mapValue,
+		parent: valuer,
+	}, key, opts, fullName)
 }
```
Comment (chenquan) [inline core/mapping/unmarshaler.go:371]: Maybe there is duplication of code?

File: core/mapping/unmarshaler.go [hunk 2]
```diff
@@ -384,7 +394,10 @@ func (u *Unmarshaler) processNamedFieldWithValue(field reflect.StructField, valu
 	fieldKind := Deref(field.Type).Kind()
 	switch fieldKind {
 	case reflect.Array, reflect.Map, reflect.Slice, reflect.Struct:
-		return u.processFieldNotFromString(field, value, mapValue, opts, fullName)
+		return u.processFieldNotFromString(field, value, valueWithParent{
+			value:  mapValue,
+			parent: vp.parent,
+		}, opts, fullName)
```
Comment (chenquan) [inline core/mapping/unmarshaler.go:400]: Maybe can be used directly the vp variables?

File: core/mapping/unmarshaler.go [hunk 3]
```diff
@@ -403,7 +416,10 @@ func (u *Unmarshaler) processNamedFieldWithValue(field reflect.StructField, valu
 			return fillPrimitive(field.Type, value, mapValue, opts, fullName)
 		}
 
-		return u.processFieldNotFromString(field, value, mapValue, opts, fullName)
+		return u.processFieldNotFromString(field, value, valueWithParent{
+			value:  mapValue,
+			parent: vp.parent,
+		}, opts, fullName)
```
Comment (chenquan) [inline core/mapping/unmarshaler.go:422]: Maybe can be used directly the vp variables?

File: core/mapping/unmarshaler.go (hunk 1 above)
Comment (kevwan) [reply core/mapping/unmarshaler.go]: good catch, and fixed. thanks!

File: core/mapping/unmarshaler.go (hunk 2 above)
Comment (kevwan) [reply core/mapping/unmarshaler.go]: fixed.

File: core/mapping/unmarshaler.go (hunk 3 above)
Comment (kevwan) [reply core/mapping/unmarshaler.go]: fixed.

File: core/mapping/valuer_test.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	valuer := recursiveValuer{
+		current: mapValuer(input["component"].(map[string]interface{})),
+		parent: simpleValuer{
+			current: mapValuer(input),
+			parent:  nil,
+		},
+	}
+	val, ok := valuer.Value("discovery")
+	fmt.Println(val, ok)
```
Comment (chenquan) [inline core/mapping/valuer_test.go:26]: Lost assert.

Comment (kevwan) [reply core/mapping/valuer_test.go]: fixed.

**Review by chenquan — APPROVED** [review]
LGTM

**Review by zhoushuguang — APPROVED** [review]

**Review by MarkJoyMa — APPROVED** [review]
LGTM

**Review by MarkJoyMa — APPROVED** [review]

**Review by chenquan — APPROVED** [review]
LGTM

**Pull request merged**